main_test.go

```diff
@@ -0,0 +1,291 @@
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Session-Token", "test-x-session-token")
+	cookie := &http.Cookie{
+		Name:   "ory_kratos_session",
+		Value:  "test-token",
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie)
+
+	//create response
+	w := httptest.NewRecorder()
+	//start function
+	handleUpdateFlow(w, req)
+	//check results
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	loginUpdateResponse := new(testLoginUpdate.LoginUpdateResponse)
+	if err := json.Unmarshal(data, loginUpdateResponse); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if loginUpdateResponse.Session_token != "test-token" {
+		t.Errorf("expected test-token, got %v", string(data))
+	}
+	if loginUpdateResponse.Session.Id != "test-1111" {
+		t.Errorf("expected test-1111, got %v", string(data))
+	}
+}
+
+func TestHandleKratosError(t *testing.T) {
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+	req := httptest.NewRequest(http.MethodGet, "/api/kratos/self-service/errors?id=1111", nil)
+	w := httptest.NewRecorder()
+	handleKratosError(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	te := new(testErrors.TestErrorReport)
+	if err := json.Unmarshal(data, te); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if te.Error.Message != "This is a test" {
+		t.Errorf("expected This is a test, got %v", string(data))
+	}
+}
+
+func TestHandleConsent(t *testing.T) {
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+	req := httptest.NewRequest(http.MethodGet, "/api/consent?consent_challenge=test_challange", nil)
+	w := httptest.NewRecorder()
+	handleConsent(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	responseRedirect := new(testConsent.OAuth2ConsentAcceptResponse)
+	if err := json.Unmarshal(data, responseRedirect); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if responseRedirect.Redirect_to != "test.test" {
+		t.Errorf("expected test.test, got %v", string(data))
+	}
+}
+
+// --------------------------------------------
+// TESTING WITH TIMEOUT SERVERS
+// currently only prints out results main.go needs pr to handle timeouts
+// --------------------------------------------
+func TestHandleCreateFlowTimeout(t *testing.T) {
+	data, err := CreateGenericTest(testServers.CreateTimeoutServers, http.MethodPut,
+		"/api/kratos/self-service/login/browser?aal=aal1&login_challenge=&refresh=false&return_to=http://test.test",
+		nil, handleCreateFlow)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	t.Logf("Result:\n%s\n", string(data))
```

Comment:
We are not asserting something in those tests. We don't gracefully handle server errors in the server yet, but if the Kratos/Hydra request times out we should just return a `500` HTTP response with some explanatory error message (for now).

Let's leave a comment about it (like `TODO: Validate response when server error handling is implemented) here.

Reply:
If it's ok with you, I'd leave the tests as they are. I left comments in main.go, and I'll implement them at some other point.